### ISSUE TITLE
iOS Extension build failed AOT container app (-O=-float32) and the extension (-O=float32).

### ISSUE THREAD

**Issue by Didi3333** [xamarin/xamarin-macios#17477]
### Steps to Reproduce

1. Create Application with single display 
2. Add Extension Broadcast Upload
3. Build solution

### Expected Behavior
Have a build which doesn't disable extensions

### Actual Behavior

MTOUCH : warning MT0113: Native code sharing has been disabled for the extension 'ScreenSharing' because the other arguments to the AOT compiler are different between the container app (-O=-float32) and the extension (-O=float32).

### Environment

<details>
<summary>Version information</summary>

```
Visual Studio Community 2022 for Mac
Version 17.4.4 (build 12)
Installation UUID: ea5f9c15-6959-4542-9e7f-0213fd556451

Runtime
.NET 6.0.12 (64-bit)
Architecture: X64

Roslyn (Language Service)
4.4.0-6.22578.12+3c6ab8e1715e5b080fb7bb77070810ab71e09387

NuGet
Version : 6.3.1.1

Kit de développement logiciel (SDK) .NET (x64)
SDK : /usr/local/share/dotnet/sdk/7.0.102/Sdks
Versions du SDK :
	7.0.102
	6.0.405
	6.0.401
	3.1.426
	3.1.423
SDK MSBuild : /Applications/Visual Studio.app/Contents/MonoBundle/MSBuild/Current/bin/Sdks

Runtime .NET (x64)
Runtime : /usr/local/share/dotnet/dotnet
Versions du runtime :
	7.0.2
	6.0.13
	6.0.9
	3.1.32
	3.1.29

Xamarin.Profiler
Version : 1.8.0.19
Emplacement : /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version : 11

Xamarin.Android
Non installé

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.12
Android Designer code EPL disponible ici :
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer code EPL disponible ici :
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.4.0.54
Hash: 6eabb9e
Branch: remotes/origin/d17-4
Build date: 2023-01-24 23:12:25 UTC

Android Device Manager
Version: 0.0.0.1206
Hash: 886af39
Branch: 886af39
Build date: 2023-01-24 23:12:25 UTC

Apple Developer Tools
Xcode 14.1 (21534.1)
Build 14B47b

Xamarin.Mac
Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

Xamarin.iOS
Version: 16.1.1.27 (Visual Studio Community)
Hash: 933c6c2c9
Branch: xcode14.1
Build date: 2022-11-22 02:00:37-0500

Xamarin Designer
Version: 17.4.0.136
Hash: d49c9ff6d3
Branch: remotes/origin/d17-4
Build date: 2023-01-24 23:12:21 UTC

Build Information
Release ID: 1704040012
Git revision: 5690bbb844f118ae3d5a7c1a945eab0f4f5f1e4c
Build date: 2023-01-24 23:10:14+00
Build branch: release-17.4
Build lane: release-17.4

Operating System
Mac OS X 13.2.0
Darwin 22.3.0 Darwin Kernel Version 22.3.0
    Thu Jan  5 20:53:49 PST 2023
    root:xnu-8792.81.2~2/RELEASE_X86_64 x86_64


```

</details>

### Build Logs

MTOUCH : warning MT0113: Native code sharing has been disabled for the extension 'ScreenSharing' because the other arguments to the AOT compiler are different between the container app (-O=-float32) and the extension (-O=float32).
MTOUCH : warning MT0113: Native code sharing has been disabled for the extension 'ScreenSharingUI' because the other arguments to the AOT compiler are different between the container app (-O=-float32) and the extension (-O=float32).

**Comment by rolfbjarne** [comment]
> Have a build which doesn't disable extensions

The warning doesn't mean the extension is disabled, just that some code sharing between the main app and the extension is disabled (so the app might be a bit bigger than necessary). Everything should work just fine otherwise.

If you have a problem with your extension, please describe the problem and we'll also need a complete test solution we can use to reproduce the problem.

**Comment by Didi3333** [comment]
When I try to debug my extension I have the following error:

warning HE0003: Could not load the framework 'DVTDeveloperModeHelper' (path: /Users/lamiscarred/Downloads/Xcode 14_1.app/Contents/SharedFrameworks/DVTDeveloperModeHelper.framework/DVTDeveloperModeHelper): not found.

**Comment by rolfbjarne** [comment]
> warning HE0003:

That's a warning, not an error.

**Comment by Didi3333** [comment]
I understand that are warning. 

I created from add project an Extension Broadcast Upload to my current project

I modified the Info.plist as com.apple.broadcast-services-upload and my extension is well visible in the control center.

When i select it and click start stream i have the  3.2.1 and after nothing, i tred to add log to the MovieClipHandler.cs and SampleHandler.cs but none appear. I'm also not able to debug extension with Visual Studio
